data_steward/tools/delete_stale_test_datasets.py

```diff
@@ -0,0 +1,88 @@
+"""
+Delete stale datasets in test environment. 
+"""
+from datetime import datetime, timezone
+import logging
+import os
+
+# Project imports
+from utils import bq, pipeline_logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _check_project(bq_client):
+    """
+    Check if the project is set to test.
+
+    :param bq_client: Client
+    :return: None if the project is set properly
+    :raise: ValueError if project is not 'aou-res-curation-test'
+    """
+    test_project = 'aou-res-curation-test'
+
+    if bq_client.project != test_project:
+        raise ValueError(
+            f'Wrong project: {bq_client.project}. This script runs only for {test_project}'
+        )
+
+    return None
+
+
+def _filter_stale_datasets(bq_client, first_n: int = None):
+    """
+    Get the first n datasets that meet all of the following criteria:
+    1. Datasets older than 90 days
+    2. Empty datasets
+
+    :param bq_client: Client
+    :param first_n: number of datasets to return. If not specified, return all.
+    :return: list of dataset names that are stale
+    """
+
+    stale_datasets, n = [], 0
+    now = datetime.now(timezone.utc)
+
+    dataset_names = [
+        dataset.dataset_id for dataset in bq_client.list_datasets()
+    ]
+
+    for dataset_name in dataset_names:
+
+        if first_n and n >= first_n:
+            break
+
+        dataset_created = bq_client.get_dataset(dataset_name).created
+
+        if (now - dataset_created).days <= 90:
+            continue
```

Comment:
I know the ticket mentions this and you are doing as it directs, but there are some things that we might have missed while grooming it. I think creation time may not be the most accurate way to identify stale datasets. I have some test datasets that I use occasionally but they were created a long time ago and I delete any tables after using them. It would be a false positive under these criteria. Let's find a time to discuss this tomorrow.

Reply:
Thanks for your review, @nishanthpp93 ! Tomorrow I will be offline here and there, so let me set up a meeting on Monday. Regarding your concern on this, does it also apply to stale buckets? The PR for stale buckets is this one: 
https://github.com/all-of-us/curation/pull/1025

Reply:
I'm not sure, let's discuss in the meeting.